// Copyright 2025 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package google.confidential_match;

import "protos/core/data_value.proto";

// A wrapper proto around DataValue used internally by the data table
// implementation.
// This contains additional metadata fields tracked along with the data.
message DataValueInternal {
  // The DataValue object containing the data stored within the object.
  DataValue data_value = 1;

  // A version identifier that ties the data value to a particular data export.
  // This is used to mark and identify which data values are kept or deleted
  // during a data refresh operation.
  uint32 version_id = 2;
}